backend/src/services/common/MeanTimeToRecovery.ts

```diff
@@ -0,0 +1,50 @@
+import { DeployInfo, DeployTimes } from "../../models/pipeline/DeployTimes";
+import {
+  AvgMeanTimeToRecovery,
+  MeanTimeToRecovery,
+  MeanTimeToRecoveryOfPipeline,
+} from "../../contract/GenerateReporter/GenerateReporterResponse";
+
+export function calculateMeanTimeToRecovery(
+  deployTimes: DeployTimes[]
+): MeanTimeToRecovery {
+  const meanTimeRecoveryPipelines: MeanTimeToRecoveryOfPipeline[] = deployTimes.map(
+    (item) => {
+      const totalTimeToRecovery = item.failed.reduce(
+        (prev: number, failedJob: DeployInfo): number => {
+          const failedJobCreateTime = new Date(
+            failedJob.pipelineCreateTime
+          ).getTime();
+          let timeToRecovery = Infinity;
+          for (let i = 0; i < item.passed.length; i++) {
```

Comment:
use foreach is better than for loop

Reply:
and about this part, I recommend that we extract to a new function to calc the timeToRecovery